src/BudgetManager/BudgetManager.CQRS/Handlers/TransactionHandlers/GetRecentTransactionsHandler.cs

```diff
@@ -0,0 +1,56 @@
+﻿using BudgetManager.CQRS.Queries.TransactionQueries;
+using BudgetManager.CQRS.Responses.TransactionResponses;
+using BudgetManager.DataAccess.MongoDbAccess.Interfaces;
+using BudgetManager.Model;
+using BudgetManager.Shared.DataAccess.MongoDB.BaseImplementation;
+using MediatR;
+using MongoDB.Driver;
+
+namespace BudgetManager.CQRS.Handlers.TransactionHandlers
+{
+    public class GetRecentTransactionsHandler : IRequestHandler<GetRecentTransactionsQuery, List<RecentTransactionsHomepageResponse>>
+    {
+        private readonly ITransactionRepository _transactionRepository;
+        private readonly IBaseRepository<User> _userRepository;
+
+        public GetRecentTransactionsHandler(ITransactionRepository transactionRepository, IBaseRepository<User> userRepository)
+        {
+            _transactionRepository = transactionRepository;
+            _userRepository = userRepository;
+        }
+
+        public async Task<List<RecentTransactionsHomepageResponse>> Handle(GetRecentTransactionsQuery request, CancellationToken cancellationToken)
+        {
+            User user = await _userRepository.FindByIdAsync(request.userId, cancellationToken);
+
+            if (user is null)
+            {
+                throw new KeyNotFoundException("User not found");
+            }
+
+            var transactions = await _transactionRepository.GetListByUserIdAsync(user.Id, cancellationToken);
+            var tenRecents = transactions.OrderByDescending(t => t.DateOfTransaction).Take(10);
```

Comment:
it is not a good way
In such situation you return all transactions of the user and then take 10 of them..

You need get only 10 transactions from db

Reply:
Also here you can use GetPageListAsync method from Transaction repository with pageNum = 1 and PageSize=10
But it's also not a good way
May be you need to create your own method to get only 10 transactions from db
```suggestion
            public async Task<IEnumerable<Transaction>> GetSeveralTransactions(FilterDefinition<Transaction> filterDefinition,
            SortDefinition<Transaction> sortDefinition,  int count, CancellationToken cancellationToken)
            {
            }
```

Reply:
Did some code changes